src/dotnet/ToolManifest/ToolManifestFile.cs

```diff
@@ -209,19 +258,60 @@ private SerializableLocalToolsManifest DeserializeLocalToolsManifest(FilePath po
             return result;
         }
 
-        private IEnumerable<(FilePath manifestfile, DirectoryPath manifestFileFirstEffectDirectory)> EnumerateDefaultAllPossibleManifests()
+        private IEnumerable<(FilePath manifestfile, DirectoryPath manifestFileFirstEffectDirectory)>
+            EnumerateDefaultAllPossibleManifests()
         {
             DirectoryPath? currentSearchDirectory = _probeStart;
             while (currentSearchDirectory.HasValue)
             {
-                var currentSearchDotConfigDirectory = currentSearchDirectory.Value.WithSubDirectories(Constants.DotConfigDirectoryName);
-                var tryManifest = currentSearchDirectory.Value.WithFile(_manifestFilenameConvention);
-                yield return (currentSearchDotConfigDirectory.WithFile(_manifestFilenameConvention), currentSearchDirectory.Value);
+                var currentSearchDotConfigDirectory =
+                    currentSearchDirectory.Value.WithSubDirectories(Constants.DotConfigDirectoryName);
+                var tryManifest = currentSearchDirectory.Value.WithFile(ManifestFilenameConvention);
+                yield return (currentSearchDotConfigDirectory.WithFile(ManifestFilenameConvention),
+                    currentSearchDirectory.Value);
                 yield return (tryManifest, currentSearchDirectory.Value);
                 currentSearchDirectory = currentSearchDirectory.Value.GetParentPathNullable();
             }
         }
 
+        private class SerializableLocalToolSinglePackage
+        {
+            public string version { get; set; }
+            public string[] commands { get; set; }
+        }
+
+        public FilePath FindFirst()
+        {
+            foreach ((FilePath possibleManifest, DirectoryPath _) in EnumerateDefaultAllPossibleManifests())
+            {
+                if (_fileSystem.File.Exists(possibleManifest.Value))
+                {
+                    return possibleManifest;
+                }
+            }
+
+            throw new ToolManifestCannotBeFoundException(
+                string.Format(LocalizableStrings.CannotFindAnyManifestsFileSearched,
+                    string.Join(Environment.NewLine,
+                        EnumerateDefaultAllPossibleManifests().Select(f => f.manifestfile.Value))));
+        }
+
+        private static bool CommandNamesEqual(ToolCommandName[] left, ToolCommandName[] right)
+        {
+            if (left == null && right == null)
```

Comment:
Nit: this can be done with fewer comparisons.

``` c#
if (left == null) return right == null;
if (right == null) return false;
return left.SequenceEqual(right);
```

Reply:
fixed